sway-core/src/parse_tree/call_path.rs

```diff
@@ -5,16 +5,17 @@ use crate::Ident;
 use sway_types::{span::Span, Spanned};
 
 /// in the expression `a::b::c()`, `a` and `b` are the prefixes and `c` is the suffix.
+/// `c` can be any type `T`, but in practice `c` is either an `Ident` or a `TypeInfo`.
 #[derive(Debug, Clone, Eq, PartialEq, Hash)]
-pub struct CallPath {
+pub struct CallPath<T> {
```

Comment:
I wonder if we could specify `Ident` as a default type here in the case that one isn't specified? E.g.

```suggestion
pub struct CallPath<T = Ident> {
```

This might help to easy the common case and cut down the diff a little?

Reply:
Good idea!